### PR TITLE
:bento: [patch] Active promotion exporter improvement

### PR DESCRIPTION
**Pull request by sunny299**
**What's the change or impact of this PR?**:
  - Set active promotion metric for prometheus

**Related issue link**:

### CODE REVIEW

File: internal/samsahai/exporter/metric.go [hunk 1]
```diff
@@ -118,42 +119,46 @@ func SetActivePromotionMetric(activeProm *s2hv1beta1.ActivePromotion) {
 	if atpState != "" {
 		switch atpState {
 		case s2hv1beta1.ActivePromotionWaiting:
-			activePromStateList["waiting"] = 1
+			activePromStateList["waiting"] = float64(time.Now().Unix())
```
Comment (zinref) [inline internal/samsahai/exporter/metric.go:122]: Could we rename variable from `activePromStateList` to `atpStateList` instead to be consistence with other

File: internal/samsahai/exporter/metric.go [hunk 2]
```diff
@@ -118,42 +119,46 @@ func SetActivePromotionMetric(activeProm *s2hv1beta1.ActivePromotion) {
...
 					state).Set(val)
 			}
-		case s2hv1beta1.ActivePromotionDestroyingPreActive, s2hv1beta1.ActivePromotionDestroyingPreviousActive,
-			s2hv1beta1.ActivePromotionFinished:
-			activePromStateList["destroying"] = 1
+		case s2hv1beta1.ActivePromotionDestroyingPreActive, s2hv1beta1.ActivePromotionDestroyingPreviousActive:
+			activePromStateList["destroying"] = float64(time.Now().Unix())
+			for state, val := range activePromStateList {
+				ActivePromotionMetric.WithLabelValues(
+					activeProm.Name,
```
Comment (zinref) [inline internal/samsahai/exporter/metric.go:153]: I think `activeProm` could be rename to `atpComp`.

File: internal/samsahai/exporter/metric.go (hunk 1 above)
Comment (Pohfy123) [inline internal/samsahai/exporter/metric.go]: could we define state type and init const for each state e.g.

```
type ActivePromotionMetricState string (I know you can find a better name than me 😂 )
const (
  ActivePromotionStateWaiting ActivePromotionMetricState = "waiting"
  ...
)
```

File: config/chart/samsahai/values.yaml [hunk 3]
```diff
@@ -67,7 +67,7 @@ service:
   http: 8080
 
 exporter:
-  enabled: false
+  enabled: true
```
Comment (Pohfy123) [inline config/chart/samsahai/values.yaml:70]: should we default at false?

File: internal/samsahai/exporter/metric.go [hunk 4]
```diff
@@ -10,6 +11,16 @@ import (
...
+type ActivePromotionMetricState string
+
+const (
+	Waiting ActivePromotionMetricState = "waiting"
+	Deploying ActivePromotionMetricState = "deploying"
+	Testing ActivePromotionMetricState = "testing"
+	Promoting ActivePromotionMetricState = "promoting"
+	Destroying ActivePromotionMetricState = "dsetroying"
+
+)
```
Comment (Pohfy123) [inline internal/samsahai/exporter/metric.go:23]: if it's being used in this package only, should we use lower case to be private?

**Pull request merged**